### PR TITLE
Ddp 6635 cancers picklist groups added to Basil

### PR DESCRIPTION
**Pull request by evgeniipr**
## Context

Update Basil configuration: added a grouped list of cancers (basically copied from feature branch `prequal-cancer-groups`, PR: https://github.com/broadinstitute/ddp-study-server/pull/905 , created by @ssettipalli).
So, this list is not the latest one yet, but it is ok for testing a frontend component (required by @aglushko-broad ).

It needs to run this configuration to dev using these scripts:
1) Invalidate last Basil study:
```
java -Dconfig.file=./output-config/application.conf -jar ./target/StudyBuilder.jar basil --vars ./output-config/basil-vars.conf ./studies/basil/study.conf --substitutions ./studies/basil/subs.conf --invalidate
```

2) Deploy a new version of Basil study:
```
java -Dconfig.file=./output-config/application.conf -jar ./target/StudyBuilder.jar basil --vars ./output-config/basil-vars.conf ./studies/basil/study.conf --substitutions ./studies/basil/subs.conf
```

### CODE REVIEW

File: study-builder/studies/basil/i18n/en.conf [hunk 1]
```diff
@@ -256,242 +256,223 @@
...
+    "optic_nerve_glioma": "Optic nerve glioma",
+    "pilocytic_astrocytoma": "Pilocytic astrocytoma (JPA)",
+    "pineal_tumor": "Pineal tumor",
+    "pituitary_tumor": "Pituitary tumor",
+    "pleomorphic_xanthoastrocytomas": "Pleomorphic xanthoastrocytomas (PXA)",
+    "primary_central_nervous_system_lymphoma": "Primary central nervous system lymphoma (PCNSL; CNS)",
+    "primitive_neuroectodermal_tumor": "Primitive neuroectodermal tumor (PNET)",
+    "schwannoma": "Schwannoma",
+    "group_prompt__breast_cancers": "Breast cancers",
+    "breast_cancer": "Breast Cancer (TNBC; MBC; IBC; ILC; IDLC; SBC; ICC)",
```
Comment (MocanaAtBroad) [inline study-builder/studies/basil/i18n/en.conf]: Not seeing my previous comment. Let me try again:
An example of how I am expecting list option labels to look like:
Breast cancer
Breast adenocarcinoma
Triple negative breast cancer (TNBC)
Metastatic breast cancer (MBC)

Comment (MocanaAtBroad) [reply study-builder/studies/basil/i18n/en.conf]: I don't think the spreadsheet is super clear about separating acronyms (abbreviations) from aliases, but not too difficult to figure out by reading the names what abbreviation goes with which alias.

Comment (evgeniipr) [reply study-builder/studies/basil/i18n/en.conf]: @MocanaAtBroad so it should be added to this breast cancer group all these options?

Comment (evgeniipr) [reply study-builder/studies/basil/i18n/en.conf]: Actually it's not quite clear to me how to interpret this content of the 2nd column for breast cancer:
```
breast adenocarcinoma; TNBC; triple negative breast cancer; metastatic breast cancer; MBC; breast carcinoma; Inflammatory Breast Cancer; IBC; Invasive Ductal Carcinoma; IDC; Invasive lobular carcinoma; ILC; mixed IDLC; IDLC; invasive mammary carcinoma; infiltrating mammary carcinoma; Metaplastic carcinoma of the breast; metaplastic; lobular; Paget's Disease of the Breast; mammary Paget's disease; Paget's disease of the nipple; Paget's disease of the nipple and areola; Secretory breast carcinoma; SBC; juvenille breast carcinoma; Tubular carcinoma; Medullary carcinoma; Invasive cribriform carcinoma; ICC; male breast cancer
```
For example, `breast carcinoma` and `Inflammatory Breast Cancer` - should be skipped?  But options `IBC; Invasive Ductal Carcinoma; IDC; Invasive lobular carcinoma` etc - should be added?

It not a problem - I can update input file and generate cancer list conf files again. Just need to understand what should be in the options' lists..

**Comment by MocanaAtBroad** [comment]
Did not find one, but make sure there is one option that has an "allowDetails": false . Needed to allow users to enter something not in list.

File: study-builder/studies/basil/i18n/en.conf [hunk 2]
```diff
@@ -256,242 +256,255 @@
...
-    "sinonasal_undifferentiated_carcinoma_snuc": "Sinonasal undifferentiated carcinoma (SNUC)",
-    "stewart_treves_syndrome": "Stewart Treves syndrome ",
-    "thyroid_cancer_carcinoma": "Thyroid cancer (carcinoma) ",
-    "von_hippellindau_disease_vhl": "Von hippel-lindau disease (VHL)",
-    "other_cancer": "Other cancer",
-    "other_cancer_details": "Other cancer details",
+    "cancer_group__brain__brain_or_cns_tumors": "Brain or Central Nervous System tumors",
+    "cancer__brain__atrt__atypical_teratoid_tumor": "Atypical teratoid rhabdoid tumor (ATRT)",
+    "cancer__brain__anapl_astrocytoma": "Anaplastic astrocytoma",
+    "cancer__brain__astrocytoma_other": "Astrocytoma, other / Cerebellar astrocytoma / Cerebral astrocytoma",
```
Comment (MocanaAtBroad) [inline study-builder/studies/basil/i18n/en.conf]: Let's make "Cerbellar astrocytoma" and "Cerebral astrocytoma" separate and indvidual options.
Looks like we are missing the standalone "Astrocytoma".
Keep "Astrocytoma, other" (I will see if they are willing to remove from list but keep it for now)

Comment (evgeniipr) [reply study-builder/studies/basil/i18n/en.conf]: Updated the cancer list

**Comment by MocanaAtBroad** [comment]
The list looks good! But I am not seeing an option with with allowDetails=true flag. Needed for user to be able to enter a cancer that is not in the list. Is one in there and I am not seeing it? https://broadinstitute.atlassian.net/browse/DDP-6281

**Comment by evgeniipr** [comment]
> The list looks good! But I am not seeing an option with with allowDetails=true flag. Needed for user to be able to enter a cancer that is not in the list. Is one in there and I am not seeing it? https://broadinstitute.atlassian.net/browse/DDP-6281

@MocanaAtBroad I added detail option

File: study-builder/studies/basil/i18n/en.conf [hunk 3]
```diff
@@ -638,6 +638,8 @@
     "cancer_additional_primary_bone_cancer": "Primary bone cancer",
     "cancer_additional_rm_retinoblastoma": "Retinoblastoma (RB)",
     "cancer_additional_vhl_von_hippel_lindau": "Von hippel-lindau disease (VHL)",
+    "other_cancer": "[es] Other cancer",
+    "other_cancer_details": "[es] Other cancer details",
```
Comment (MocanaAtBroad) [inline study-builder/studies/basil/i18n/en.conf]: Not a big deal because this option is invisible, but don't need [es] prefix.

Comment (evgeniipr) [reply study-builder/studies/basil/i18n/en.conf]: Removed [es] prefixes

Comment (MocanaAtBroad) [reply study-builder/studies/basil/i18n/en.conf]: Please add all the es prefixes again to the es.conf file. They were not necessary in this specific location/file. We don't use them in the English file, only in the Spanish file to highlight text that we don't have a translation for. The requested change was just for these two lines.

Comment (evgeniipr) [reply study-builder/studies/basil/i18n/en.conf]: Re-fixed

File: study-builder/studies/basil/snippets/question-primary-cancer-groups-self.conf [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      "stableId": "OTHER_CANCER",
+      "optionLabelTemplate": {
+        "templateType": "TEXT",
+        "templateText": "$cancer_other",
+        "variables": [
+          {
+            "name": "cancer_other",
+            "translations": [
+              { "language": "en", "text": ${i18n.en.cancer.other_cancer}},
+              {"language": "es", "text": ${i18n.es.cancer.other_cancer}}
```
Comment (MocanaAtBroad) [inline study-builder/studies/basil/snippets/question-primary-cancer-groups-self.conf:39]: With the allowDetails=true option, this option is not needed.

**Review by MocanaAtBroad — DISMISSED** [review]

**Review by MocanaAtBroad — APPROVED** [review]
